### PR TITLE
fix upload bug found by brayoh

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/actions/fs/index.js [hunk 1]
```diff
@@ -619,6 +619,13 @@ function* openPathItem(action: FsGen.OpenPathItemPayload): Saga.SagaGenerator<an
   )
 }
 
+const uploadFanOut = (state: TypedState, action: FsGen.UploadsPayload) =>
+  Saga.sequentially(
```
Comment (chrisnojima) [inline shared/actions/fs/index.js]: why sequentially vs all?

File: shared/actions/fs/index.js [hunk 2]
```diff
@@ -619,6 +619,13 @@ function* openPathItem(action: FsGen.OpenPathItemPayload): Saga.SagaGenerator<an
...
+    action.payload.localPaths.map(localPath =>
```
Comment (chrisnojima) [inline shared/actions/fs/index.js]: why do this, why not just handle it natively instead of converting it ?

Comment (songgao) [reply shared/actions/fs/index.js]: You mean the existing `upload` handler should just take a `localPaths` array instead of just a single `localPath`?

Comment (songgao) [reply shared/actions/fs/index.js]: Each individual upload would be a separate RPC call though, so I figured doing a fan out here is cleaner.

File: shared/actions/fs/index.js (hunk 1 above)
Comment (songgao) [reply shared/actions/fs/index.js]: I didn't pick sequentially over all consciously. I guess I just didn't use `all` before and assumed `sequentially` was preferred. Do we prefer `all` when order doesn't matter?

Comment (chrisnojima) [reply shared/actions/fs/index.js]: yes, do things in parallel

File: shared/actions/fs/index.js (hunk 2 above)
Comment (chrisnojima) [reply shared/actions/fs/index.js]: then why bother with this action at all, just emit the individual ones upstream

Comment (songgao) [reply shared/actions/fs/index.js]: I was trying to keep actionToPromise there (see the changes in `shared/actions/fs/platform-specific.desktop.js` below). If we need all these multiple actions, it seems like we'd need to use something like `takeEvery` there?

**Comment by songgao** [comment]
bumpy ping!

**Pull request merged**